zebra-rpc/src/server.rs

```diff
@@ -47,35 +45,22 @@ pub struct RpcServer;
 
 impl RpcServer {
     /// Start a new RPC server endpoint
-    pub fn spawn<Version, Mempool, State, Tip>(
+    pub fn spawn<Version, Mempool, State, Tip, BlockVerifier>(
         config: Config,
         app_version: Version,
         mempool: Buffer<Mempool, mempool::Request>,
         state: State,
-        // TODO: use cfg-if to apply trait constraints behind feature flag and remove the `Option`.
-        #[cfg(feature = "getblocktemplate-rpcs")] block_verifier: Option<
-            BlockVerifier<
-                Buffer<
-                    BoxService<
-                        zebra_state::Request,
-                        zebra_state::Response,
-                        Box<dyn std::error::Error + Send + Sync>,
-                    >,
-                    zebra_state::Request,
-                >,
-                Buffer<
-                    BoxService<transaction::Request, transaction::Response, TransactionError>,
-                    transaction::Request,
-                >,
-            >,
-        >,
+        #[allow(unused_variables)] block_verifier: BlockVerifier,
```

Comment:
It seemed simpler to always accept this argument, and then handle it being optional within `zebra-rpc`.